code/datums/components/chewable.dm

```diff
@@ -0,0 +1,80 @@
+/// Anything with this component will provide the reagents inside the
+/// item to the user when it is equipped.
+/datum/component/chewable
+	/// A bitfield of valid slots. If this is not provided, then it will
+	/// use the `slot_flags` of the item.
+	var/slots_to_check
+
+	/// The time left before it's deleted.
+	var/lifetime = 6 MINUTES
```

Comment:
You may notice the older value is 360, which is not 6 minutes (which == 3600). This is because the old system was in seconds, this one is in deciseconds. I really don't like that delta_time is in seconds, but here we are.

Remember this for other values, such as `15 * 60` turning into `15 MINUTES`. The behavior is the same, it's just that delta_time is being multiplied by 10 now.

Reply:
the concept of a "timed" food is also extremely cringe because it's made of reagents which already are a measure of volume, an abstract lifetime doesn't really make much sense and was made by someone who chews on penis.

Reply:
This exists on cigarettes for the same reasons and it's because to have these items visibly last a decent amount of time they'd have to be filled with hundreds of units of reagents and microdosing is otherwise very buggy and unreliable, so the reagent transfer rate can't be too low to match a longer duration

Not saying it's a good way to handle it but it exists for a reason and nobody came up with a better solution